llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/ConvertConv2DToWinograd.cpp

```diff
@@ -125,6 +130,7 @@ class FoldWinogradFilterTransform final
     ShapedType type = constOp->getResult(0).getType().cast<ShapedType>();
     Type elementType = type.getElementType();
     assert(elementType.isa<FloatType>());
+    auto floatType = elementType.cast<FloatType>();
```

Comment:
The elementType is only used by the cast. And the `cast` has assertion in the implementation. How about rewrite it into a single statement. E.g.,

```cpp
auto elemType = type.getElementType().cast<FloatType>();

Reply:
makes sense. With the assert inside the case, we don't need the explicit case. Will remove.